### PR TITLE
Fixed edit user profile

### PR DESCRIPTION
**Pull request by Starhkz**

### CODE REVIEW

File: lib/services/current_user_profile.dart [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    String endPoint = '/organizations/$orgId/members?query=$userEmail';
+    print(
+        'The email is ${storageService.getString(StorageKeys.currentUserEmail)}');
+    //https://api.zuri.chat/organizations/614b53f244a9bd81cedc0bc9/members?query=jbadmin@gmail.com
+
+    final res = await _api.get(endPoint, headers: {
+      'Authorization':
+          'Bearer ${storageService.getString(StorageKeys.currentSessionToken)}'
+    });
+    final snackbar = locator<SnackbarService>();
```
Comment (isaacadariku) [inline lib/services/current_user_profile.dart:29]: Can you move this  `final snackbar = locator<SnackbarService>();` out the function and register it above.

File: lib/services/current_user_profile.dart [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  final storageService = locator<SharedPreferenceLocalStorage>();
+
+  /// Fetches info of the current user
+  Future<ProfileModel> currentUser() async {
+    String? orgId = storageService.getString(StorageKeys.currentOrgId);
+    //String? userId = storageService.getString(StorageKeys.currentUserId);
+    String? userEmail = storageService.getString(StorageKeys.currentUserEmail);
+    String endPoint = '/organizations/$orgId/members?query=$userEmail';
+    print(
```
Comment (isaacadariku) [inline lib/services/current_user_profile.dart:21]: Remove all print statement and use a logger

File: lib/services/current_user_profile.dart [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      );
+
+      snackbar.showCustomSnackBar(
+        duration: const Duration(seconds: 5),
+        variant: SnackbarType.success,
+        message: ''' Profile Loaded  Successfully'''
+            '''  Please Exit The Edit Profile Page Then Open It Once More ''',
+      );
+
+      return ProfileModel(
```
Comment (isaacadariku) [inline lib/services/current_user_profile.dart:63]: Why are you manually serializing all data.

You can get the data in a variable and use the model `fromJson` method to get all.

Please make the changes in all,

File: lib/ui/shared/bottom_sheets/profile_botton_sheet/custom_user_bottom_sheet_viewmodel.dart [hunk 4]
```diff
@@ -1,18 +1,32 @@
...
 import '../../../../app/app.locator.dart';
 import '../../../../app/app.router.dart';
 import '../../../../models/user_model.dart';
-import '../../../../package/base/server-request/api/http_api.dart';
-import '../../../../services/local_storage_services.dart';
-import '../../../../utilities/constants.dart';
-import '../../../../utilities/storage_keys.dart';
 
-class CustomUserBottomSheetViewModel extends FutureViewModel {
+final storageService = locator<SharedPreferenceLocalStorage>();
```
Comment (isaacadariku) [inline lib/ui/shared/bottom_sheets/profile_botton_sheet/custom_user_bottom_sheet_viewmodel.dart:11]: move this into the class

File: lib/ui/shared/bottom_sheets/profile_botton_sheet/custom_user_bottom_sheet_viewmodel.dart [hunk 5]
```diff
@@ -1,18 +1,32 @@
...
+class CustomUserBottomSheetViewModel extends BaseViewModel {
+  String photoUrl =
+      storageService.getString(StorageKeys.currentOrgId).toString();
+
+  String name = storageService.getString(StorageKeys.firstName).toString();
+
+  String displayName =
+      storageService.getString(StorageKeys.displayName).toString();
+
+  String phoneNum = storageService.getString(StorageKeys.phone_num).toString();
```
Comment (isaacadariku) [inline lib/ui/shared/bottom_sheets/profile_botton_sheet/custom_user_bottom_sheet_viewmodel.dart:22]: The way you are passing data to get to this view is not proper, you are making too many storage request. Why not when the view loads, it get it from the api and show it

File: lib/ui/view/view_profile_page/view_profile.dart [hunk 6]
```diff
@@ -8,8 +11,8 @@ import 'widgets/profile_action.dart';
 import 'widgets/profile_list.dart';
 
 class ViewProfile extends StatelessWidget {
- final bool isActive = true;
-
+  final bool isActive = true;
+  final storageService = locator<SharedPreferenceLocalStorage>();
```
Comment (isaacadariku) [inline lib/ui/view/view_profile_page/view_profile.dart:15]: remove any logic from the view

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Few changes and we are good to go

File: lib/ui/shared/bottom_sheets/profile_botton_sheet/custom_user_bottom_sheet_viewmodel.dart (hunk 5 above)
Comment (Starhkz) [reply lib/ui/shared/bottom_sheets/profile_botton_sheet/custom_user_bottom_sheet_viewmodel.dart:22]: > The way you are passing data to get to this view is not proper, you are making too many storage request. Why not when the view loads, it get it from the api and show it

The data is passed to the storage after login to decrease load time. The backend takes long to respond so the data is stored immediately after login

File: lib/ui/shared/bottom_sheets/profile_botton_sheet/custom_user_bottom_sheet_viewmodel.dart (hunk 4 above)
Comment (Starhkz) [reply lib/ui/shared/bottom_sheets/profile_botton_sheet/custom_user_bottom_sheet_viewmodel.dart:11]: > move this into the class

Causes an error: "The instance member 'storageService' can't be accessed in an initializer."

**Comment by Blazebrain** [comment]
Still using this?

Fix conflicts

**Comment by Starhkz** [comment]
> Still using this?
> 
> Fix conflicts

The fix is in the new PR that was closed.
Can't fix this because I had to delete repo to fetch upstream.

**Comment by YenumJay** [comment]
Pull latest changes from dev and resolve conflicts.

If this is currently not in use, close the PR

**Pull request closed without merging**